File: app/src/main/java/com/xattacker/android/test/MainActivity.java
package com.xattacker.android.test;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import com.xattacker.android.view.circular.CircularMenuLayout;
import com.xattacker.android.view.circular.CircularMenuListener;
import com.xattacker.android.view.circular.CircularMenuMode;

public class MainActivity extends AppCompatActivity implements CircularMenuListener
{
    private CircularMenuLayout _circularLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        _circularLayout = findViewById(R.id.view_circle);
        _circularLayout.setListener(this);
    }

    @Override
    public void onCircularMenuClicked(View aMenuView)
    {
        android.util.Log.d("aaa", aMenuView.toString());
    }

    public void onModeClick(View aView)
    {
        if (_circularLayout != null)
        {
            _circularLayout.setMode(CircularMenuMode.AUTO);
        }
    }

    public void onMode2Click(View aView)
    {
        if (_circularLayout != null)
        {
            _circularLayout.setMode(CircularMenuMode.MANUAL);
        }
    }
}

File: app/src/main/java/com/xattacker/android/test/MainActivity.kt
package com.xattacker.android.test

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.View
import com.xattacker.android.circularmenulayout.R
import com.xattacker.android.view.circular.CircularMenuLayout
import com.xattacker.android.view.circular.CircularMenuListener
import com.xattacker.android.view.circular.CircularMenuMode

class MainActivity : AppCompatActivity(), CircularMenuListener
{
    private var circularLayout: CircularMenuLayout? = null

    override fun onCreate(savedInstanceState: Bundle?)
    {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)

        circularLayout = findViewById(R.id.view_circle)
        circularLayout?.listener = this
    }

    override fun onCircularMenuClicked(aMenuView: View)
    {
        android.util.Log.d("aaa", aMenuView.toString())
    }

    fun onModeClick(aView: View)
    {
        circularLayout?.mode = CircularMenuMode.AUTO
    }

    fun onMode2Click(aView: View)
    {
        circularLayout?.mode = CircularMenuMode.MANUAL
    }
}
